ksql-engine/src/main/java/io/confluent/ksql/util/SchemaRegistryClientFactory.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+
+public class SchemaRegistryClientFactory {
```

Comment:
I'm not sure why we need this. If we do need it can we make it non-static?

Reply:
This represents the default schema registry service for the engine and will be used by the serde if the user session does not have a different schema registry. This needs to be singleton.

Reply:
I still don't think it needs to be static. The engine or whatever can create one and pass it to where it is needed.